internal/cert/default_cert.go

```diff
@@ -44,7 +44,7 @@ var errSourceUnavailable = errors.New("certificate source is unavailable")
 //
 // If neither source is available (due to missing configurations), a nil Source and a nil Error are
 // returned to indicate that a default certificate source is unavailable.
-func DefaultSource() (Source, error) {
+var DefaultSource = func() (Source, error) {
```

Comment:
can you explain why the change here? If it is only to make it convenient for testing, then I'd rather we duplicate some code for testing instead of affecting prod execution path. Thanks!

Reply:
this is just so that we can override it in unit tests for cases where we want no client cert to be available. 
On cloudtop for example, secureConnect is configured, so there will be cert source by default. 

let me know if you have other ideas. thanks!

Reply:
If that's the objective, you can simply override the os env var "GOOGLE_API_USE_CLIENT_CERTIFICATE" to "false" in your unit tests, and a nil cert source should be returned (make sure you restore the value to it's original value at the end of the test).